src/langmem/short_term/summarization.py

```diff
@@ -0,0 +1,206 @@
+from typing import Callable, Optional, cast
+
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_core.messages import AIMessage, BaseMessage, HumanMessage, SystemMessage
+from langchain_core.prompts.chat import ChatPromptTemplate, ChatPromptValue
+from pydantic import BaseModel
+
+TokenCounter = Callable[[list[BaseMessage]], int]
+
+
+DEFAULT_INITIAL_SUMMARY_PROMPT = ChatPromptTemplate.from_messages(
+    [
+        ("placeholder", "{messages}"),
+        ("user", "Create a summary of the conversation above:"),
+    ]
+)
+
+
+DEFAULT_EXISTING_SUMMARY_PROMPT = ChatPromptTemplate.from_messages(
+    [
+        ("placeholder", "{messages}"),
+        (
+            "user",
+            "This is summary of the conversation to date: {existing_summary}\n\n"
+            "Extend the summary by taking into account the new messages above:",
+        ),
+    ]
+)
+
+DEFAULT_FINAL_SUMMARY_PROMPT = ChatPromptTemplate.from_messages(
+    [
+        # if exists
+        ("placeholder", "{system_message}"),
+        ("system", "Summary of conversation earlier: {summary}"),
+        ("placeholder", "{messages}"),
+    ]
+)
+
+
+class SummaryInfo(BaseModel):
```

Comment:
Could maybe be a protocol so it doesn't have to be pydantic?

Reply:
yea probably wanna avoid pydantic here